File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        
        if (root ==null){
            return 0;
        }
        
        int x;
        int y;
        
        x = minDepth(root.left);
        y = minDepth(root.right);
        
        if(root.left==null){
            return y+1;
        }
        
        else if(root.right==null){
            return x+1;
        }
        
        return 1 + Math.min(x,y);
        
        
    }
}

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        
        
        # BFS
        if not root:
            return 0
        
        q = deque([(root, 1)])

        while q:
            cur, depth = q.popleft()
            if not cur.left and not cur.right:
                return depth
            
            else:
                if cur.left:
                    q.append((cur.left, depth+1))
                if cur.right:
                    q.append((cur.right, depth+1))

        
        
        
#         if not root:
#             return 0
        
#         if root.left is None and root.right is None:
#             return 1
        
#         if root.left is None:
#             return 1 + self.minDepth(root.right)
        
#         if root.right is None:
#             return 1 + self.minDepth(root.left)
        
        
#         return 1 + min(self.minDepth(root.left),self.minDepth(root.right))
            
        